File: TeachingAppAPI/Entities/Topic.cs
using System;
using System.Collections.Generic;

namespace TeachingAppAPI.Entities
{
    public partial class Topic
    {
        public Topic()
        {
            Lesson = new HashSet<Lesson>();
            Quiz = new HashSet<Quiz>();
        }

        public int TopicId { get; set; }
        public int? CourseId { get; set; }
        public string TopicName { get; set; }
        public string TopicDesc { get; set; }
        public int? TopicOrder { get; set; }

        public Course Course { get; set; }
        public ICollection<Lesson> Lesson { get; set; }
        public ICollection<Quiz> Quiz { get; set; }
    }
}

File: TeachingAppAPI/Client-App/src/app/create-course/course-content/topic.ts
import { Lesson } from "./lesson";

export class Topic{
    topicId?: number;
    topicName: string;
    topicDesc: string;
    topicOrder: number;
    topicLessons: Lesson[];

    constructor(topicId, topicName, topicDesc, topicOrder, topicLessons){
        this.topicId = topicId;
        this.topicName = topicName;
        this.topicDesc = topicDesc;
        this.topicOrder = topicOrder;
        this.topicLessons = topicLessons;
    }
}